cmd/wal/truncate/cmd.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2024 StreamNative, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package truncate
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/streamnative/oxia/server/wal"
+	"log/slog"
+	"math"
+)
+
+type truncateOptions struct {
+	namespace string
+	shard     int64
+	walDir    string
+
+	lastEntry      bool
+	safePointEntry int64
+}
+
+var (
+	options = truncateOptions{}
+	Cmd     = &cobra.Command{
+		Use:   "truncate",
+		Short: "truncate the WAL",
+		Long:  `truncate the WAL by some conditions`,
+		RunE:  exec,
+	}
+)
+
+func init() {
+	Cmd.Flags().Int64Var(&options.shard, "shard", 0, "shard id")
+	Cmd.Flags().StringVar(&options.namespace, "namespace", "default", "namespace name")
+	Cmd.Flags().StringVar(&options.walDir, "wal-dir", "", "directory path")
```

Comment:
These should be options of the `wal` command, so they get inherited by all subcommands

Reply:
applied.